test/CMakeLists.txt

```diff
@@ -1,69 +1,103 @@
-cmake_minimum_required(VERSION 3.14)
-project(NESO-Particles VERSION 0.0.1)
+#No need should be inherited from neso-particles
+#set(CMAKE_CXX_STANDARD 17)
 
-# GoogleTest requires at least C++11 DataParallel C++ requires at least C++17
-set(CMAKE_CXX_STANDARD 17)
-
-add_definitions(-DGPU_SELECTOR=0)
+#add_definitions(-DGPU_SELECTOR=0)
 
 find_package(GTest QUIET)
 
 if(NOT GTest_FOUND)
-  include(FetchContent)
-  FetchContent_Declare(
-    googletest
-    URL https://github.com/google/googletest/archive/054a986a8513149e8374fc669a5fe40117ca6b41.zip
-  )
-  # For Windows: Prevent overriding the parent project's compiler/linker
-  # settings
-  set(gtest_force_shared_crt
-      ON
-      CACHE BOOL "" FORCE)
-  FetchContent_MakeAvailable(googletest)
+    include(FetchContent)
+    FetchContent_Declare(
+        googletest
+        URL https://github.com/google/googletest/archive/054a986a8513149e8374fc669a5fe40117ca6b41.zip
+        DOWNLOAD_EXTRACT_TIMESTAMP FALSE
+        )
+    # For Windows: Prevent overriding the parent project's compiler/linker
+    # settings
+    set(gtest_force_shared_crt
+        ON
+        CACHE BOOL "" FORCE)
+    FetchContent_MakeAvailable(googletest)
 endif()
 
-set(EXECUTABLE testNESOParticles)
-set(TEST_MAIN ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp)
 
-# List test source files
-file(GLOB TEST_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/test_*.cpp
-     ${CMAKE_CURRENT_SOURCE_DIR}/*_test.cpp)
-
-# options common to both fftw/mkl
-set(LINK_LIBRARIES_COMMON
-    ${MPI_CXX_LINK_FLAGS} ${MPI_CXX_LIBRARIES} ${NESO_PARTICLES_LIBRARIES}
-    ${NESO_PARTICLES_LINK_FLAGS} GTest::gtest)
+#Find SYCL
+if (COMMAND add_sycl_to_target) 
+    #Not perfect method but if in a submodule dont want to use a different
+    #sycl
+    message("add_sycl_to_target macro exists")
+else()
+    include(${CMAKE_SOURCE_DIR}/cmake/SYCL.cmake)
+    message("add_sycl_to_target macro NOT FOUND looking for SYCL package...")
+    find_sycl()
+    if(SYCL_FOUND)
+        message(STATUS "Using SYCL implementation ${SYCL_IMPLEMENTATION}")
+        if(NOT SYCL_INTEL)
+            set(HIPSYCL_TARGETS "omp")
+            set(ACPP_TARGETS "omp")
+            set(SYCL_FLAG NESO_HIPSYCL)
+        else()
+            set(SYCL_FLAG NESO_DPCPP)
+        endif()
+    endif()
```

Comment:
I don't think we should override the configuration the user set for sycl targets. I think it's reasonable to expect the user specify this - i.e. would simply remove this block.

Reply:
I see what you mean and agree that ideally the user would just specify what they want - (see the NESO pull request for something closer (but pretty far from) my ideal).  I have tried to replicate what is happening now (but in the top level CMakeLists.txt) because I didn't want to break anybody's workflow.

If it was down to me I would just have some sane default options and get rid of all the hand-holding via macros, or find packages and replace with advice in the README?

Reply:
I'm fairly sure all of the existing workflows expect the user to specify the configuration, e.g.

```
cmake -DHIPSYCL_TARGETS=omp.library-only -DCMAKE_BUILD_TYPE=Debug .
cmake -DHIPSYCL_TARGETS=cuda-nvcxx -DCMAKE_BUILD_TYPE=Debug .
```

Reply:
I think we should assume that the sycl implementation itself has a sane default.

Reply:
AFAIKS the top level CMakeLists.txt does exactly what I am doing (line 34) - I just copied that behaviour. I will remove it though.
One concern is what if you specify your HIPSYCL_TARGETS but it finds IntelSYCL or whatever and silently ignores what you asked for - this is (I think a good) argument for not even trying to find a sycl and forcing people to specify (more like I have done in NESO - though that defaults to hunting for a suitable implementation)

Reply:
I installed hipsycl and it works for me as expected - I don't have two target platforms on here though I will check on on another system.
Stupid question but is it becaue you wrote -D*CMAKE*_TARGETS.

Reply:
Woooops, thanks.

Reply:
What about this?

- I'll get rid of the macro thing completely
- Add some option to specify a SYCL implementation (+version) to look for. 
- If it is specified and  fails to find it stop there
- If it isn't specified just warn and carry on under the assumption they know what they are doing

This will probably break something for somebody, but won't do unexpected things like ignore HIPSYCL_TARGETS that have been specified because it found some other sycl. Also it's simpler.